### ISSUE TITLE
Add `claim_cashdrop()` call to transfer alts in `currency` - Cashdrop

### ISSUE THREAD

**Issue by balqaasem** [setheum/setheum#297]
```

		/// Transfer some balance to another account under `currency_id`.
		///
		/// The dispatch origin for this call must be `Signed` by the
		/// transactor.
		#[pallet::weight(T::WeightInfo::transfer_non_native_currency())]
		pub fn transfer(
			origin: OriginFor<T>,
			dest: <T::Lookup as StaticLookup>::Source,
			currency_id: CurrencyIdOf<T>,
			#[pallet::compact] amount: BalanceOf<T>,
			// TODO: Add `claim_cashdrop: bool`, and if yes then call clain_cashdrop(origin, currency_id, amount) to claim cashdrop.
		) -> DispatchResultWithPostInfo {
			let from = ensure_signed(origin)?;
			let to = T::Lookup::lookup(dest)?;
			ensure!(
				!T::FiatCurrencyIds::get().contains(&currency_id),
				Error::<T>::InvalidCurrencyType,
			);
			<Self as MultiCurrency<T::AccountId>>::transfer(currency_id, &from, &to, amount)?;
			Ok(().into())
		}
```

**Comment by balqaasem** [comment]
Add this to precompiles and predeployed contracts too

**Comment by balqaasem** [comment]
Closed in Closes in Serp settpay #383
